recipe/install_nvcc.sh

```diff
@@ -2,61 +2,98 @@
 
 set -xeuo pipefail
 
-# Default to using `nvcc` to specify `CUDA_HOME`.
-if [ -z ${CUDA_HOME+x} ]
-then
-    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
-fi
-
 # Set `CUDA_HOME` in an activation script.
 mkdir -p "${PREFIX}/etc/conda/activate.d"
-cat > "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
+cat >"${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
 #!/bin/bash
 
-if [[ ! -d "${CUDA_HOME}" ]]
+export CUDA_HOME_CONDA_NVCC_BACKUP="\${CUDA_HOME:-}"
+export CFLAGS_CONDA_NVCC_BACKUP="\${CFLAGS:-}"
+export CPPFLAGS_CONDA_NVCC_BACKUP="\${CPPFLAGS:-}"
+export CXXFLAGS_CONDA_NVCC_BACKUP="\${CXXFLAGS:-}"
+
+# Default to using \$(cuda-gdb) to specify \$(CUDA_HOME).
+if [ -z \${CUDA_HOME+x} ]
 then
-    echo "Cannot find: ${CUDA_HOME}"
-    exit 1
+    CUDA_GDB_EXECUTABLE=\$(which cuda-gdb)
+    if [ -n "\$CUDA_GDB_EXECUTABLE" ]; then
+        CUDA_HOME=\$(dirname \$(dirname \$CUDA_GDB_EXECUTABLE))
+    else
+        echo "Cannot determine CUDA_HOME: cuda-gdb not in PATH"
+        exit 1
+    fi
 fi
-if [[ ! -f "${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
+
+if [[ ! -d "\${CUDA_HOME}" ]]
 then
-    echo "Cannot find: ${CUDA_HOME}/lib64/stubs/libcuda.so"
+    echo "Cannot find: \${CUDA_HOME}"
     exit 1
 fi
-grep "CUDA Version ${PKG_VERSION}" ${CUDA_HOME}/version.txt &>/dev/null
-if [[ $? -ne 0 ]]
+
+if [[ ! -f "\${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
 then
+    echo "Cannot find: \${CUDA_HOME}/lib64/stubs/libcuda.so"
+    exit 1
+fi
+
+if [[ \$(grep -q "CUDA Version ${PKG_VERSION}" \${CUDA_HOME}/version.txt) -ne 0 ]]; then
     echo "Version of installed CUDA didn't match package"
     exit 1
 fi
 
-export CUDA_HOME="${CUDA_HOME}"
+export CUDA_HOME="\${CUDA_HOME}"
 export CFLAGS="\${CFLAGS} -I\${CUDA_HOME}/include"
 export CPPFLAGS="\${CPPFLAGS} -I\${CUDA_HOME}/include"
 export CXXFLAGS="\${CXXFLAGS} -I\${CUDA_HOME}/include"
+
+# Add \$(libcuda.so) shared object stub to the compiler sysroot.
+# Needed for things that want to link to \$(libcuda.so).
+# Stub is used to avoid getting driver code linked into binaries.
+
+CONDA_ENV_SYSROOT="\$(\${CC} --print-sysroot)"
```

Comment:
Not sure I'm following. What is the relationship? That issue seems to be about setting `CC`, which is not what is happening here.

Reply:
Here `CC` is assumed to contain a path to a compiler (obviously) but when `nvcc` and `go` are both used as dependencies to a third package, the `go` activate will reset `CC` to nonsense (as described in https://github.com/conda-forge/go-feedstock/issues/47). If `go` activate precedes to `nvcc` activate, the `$CC --print-sysroot` will fail with a message
```
nvcc_linux-64_activate.sh: line 46: /use/the/cgo/conda/package/instead: No such file or directory
```

I realize that this is actually a `go` issue but `go` feedstock maintainers refuse to recognize this.
Hence my suggestion is to avoid using `CC` when activating `nvcc` if possible. The build script of the third-party package can workaround the `go-CC` issue (by resetting `CC`) but the workaround will not be possible if the failure happens during the activation of dependencies.

Reply:
How often are people using `go` with `nvcc`? In any event, as you say, this is an issue with the `go` feedstock.

Reply:
OmniSciDB, for instance, is using go and nvcc.
At least to me, it is a real issue.

Reply:
Still this is the wrong place to handle this. It should be fixed in the `go` package. They shouldn't be setting `CC`.

Reply:
Besides what `go` is doing will break things for us [in other places]( https://github.com/conda-forge/nvcc-feedstock/blob/5e21ee5bf0c8f2bf5829c21521f48c3cff8bc419/recipe/install_nvcc.sh#L53 ).

Reply:
Right. But at least, the current PR allows a workaround to the `go` issue for the `nvcc` script (one can reset `$CXX` before calling `nvcc`).

Reply:
Btw, any help convincing the go feedstock maintainers about this issue is appreciated.

Reply:
The [last comment]( https://github.com/conda-forge/go-feedstock/issues/47#issuecomment-512681397 ) seems to suggest they are open to changing things, but lack the bandwidth to do so.

Reply:
If we are to come up with a workaround, I'd be more inclined to see it live in a recipe that use `go` and `nvcc` than in the `nvcc` package.

Reply:
Agree with @jakirkham here that workarounds should ideally be at the intersection of `nvcc` and `go` until we can get this resolved upstream in `go`